test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -4768,6 +4768,717 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
+            {
+                // Arrange
+                var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(
+                    new List<Configuration.PackageSource>()
+                    {
+                        new Configuration.PackageSource(packageSource.Path)
+                    });
+
+                using (var testSolutionManager = new TestSolutionManager(true))
+                {
+                    var testSettings = new Configuration.NullSettings();
+                    var token = CancellationToken.None;
+                    var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                    var nuGetPackageManager = new NuGetPackageManager(
+                        sourceRepositoryProvider,
+                        testSettings,
+                        testSolutionManager,
+                        deleteOnRestartManager);
+                    var projectA = testSolutionManager.AddNewMSBuildProject("testA");
+
+                    var projectActions = new List<NuGetProjectAction>();
+                    projectActions.Add(
+                        NuGetProjectAction.CreateInstallProjectAction(
+                            new PackageIdentity("inValidPackageA", new NuGetVersion("1.0.0")),
+                            sourceRepositoryProvider.GetRepositories().First()));
+
+                    // batch handlers
+                    var batchStartIds = new List<string>();
+                    var batchEndIds = new List<string>();
+
+                    // add batch events handler
+                    nuGetPackageManager.BatchStart += (o, args) =>
+                    {
+                        batchStartIds.Add(args.Id);
+                    };
+
+                    nuGetPackageManager.BatchEnd += (o, args) =>
+                    {
+                        batchEndIds.Add(args.Id);
+                    };
+
+                    Exception exception = null;
+                    try
+                    {
+                        // Act
+                        await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(projectA, projectActions,
+                            new TestNuGetProjectContext(), token);
+                    }
+                    catch (Exception ex)
+                    {
+                        exception = ex;
+                    }
+
+                    // Assert
+                    Assert.NotNull(exception);
+
+                    // Check batch events data
+                    Assert.True(batchStartIds.Count == 1);
+                    Assert.True(batchEndIds.Count == 1);
+                    Assert.Equal(batchStartIds[0], batchEndIds[0]);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestPacMan_InstallPackage_BuildIntegratedProject_BatchEvent_NotRaised()
+        {
+            // Arrange
+            var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV3OnlySourceRepositoryProvider();
```

Comment:
I tried that but somehow it wasn't working for build integrated project so i used v3 source. I'll take another look.